wurst/_handles/Camera.wurst

```diff
@@ -0,0 +1,166 @@
+
+public function setCameraQuickPosition(vec2 pos)
+    SetCameraQuickPosition(pos.x, pos.y)
+
+public function setCameraBounds(vec2 bottomLeft, vec2 topRight, vec2 topLeft, vec2 bottomRight)
+    SetCameraBounds(bottomLeft.x, bottomLeft.y, topRight.x, topRight.y, topLeft.x, topLeft.y, bottomRight.x, bottomRight.y)
+
+public function panCameraTo(vec2 pos)
+    PanCameraTo(pos.x, pos.y)
+
+public function panCameraToTimed(vec2 pos, real duration)
+    PanCameraToTimed(pos.x, pos.y, duration)
+
+public function panCameraToWithZ(vec2 pos, real zOffsetDest)
+    PanCameraToWithZ(pos.x, pos.y, zOffsetDest)
+
+public function panCameraToTimedWithZ(vec2 pos, real zOffsetDest, real duration)
+    PanCameraToTimedWithZ(pos.x, pos.y, zOffsetDest, duration)
+
+public function setCameraRotateMode(vec2 pos, angle toSweep, real duration)
+    SetCameraRotateMode(pos.x, pos.y, toSweep.radians(), duration)
+
+public function camerafield.set(real value, real duration)
+    SetCameraField(this, value, duration)
+
+public function camerafield.adjust(real offset, real duration)
+    AdjustCameraField(this, offset, duration)
+
+public function camerafield.get() returns real
+    return GetCameraField(this)
+
+public function camerafield.getFieldSwap(camerasetup whichSetup) returns real
+    return CameraSetupGetFieldSwap(this, whichSetup)
+
+public function camerafield.setCameraField(player whichPlayer, real value, real duration)
+    SetCameraFieldForPlayer(whichPlayer, this, value, duration)
+
+public function camerasetup.setField(camerafield whichField, real value, real duration)
+    CameraSetupSetField(this, whichField, value, duration)
+
+public function camerasetup.getField(camerafield whichField) returns real
+    return CameraSetupGetField(this, whichField)
+
+public function camerasetup.setDestPosition(vec2 pos, real duration)
+    CameraSetupSetDestPosition(this, pos.x, pos.y, duration)
+
+public function camerasetup.getDestPosition() returns vec2
+    return vec2(CameraSetupGetDestPositionX(this),CameraSetupGetDestPositionY(this))
+
+public function camerasetup.apply(boolean doPan, boolean panTimed)
+    CameraSetupApply(this, doPan, panTimed)
+
+public function camerasetup.applyWithZ(real zDestOffset)
+    CameraSetupApplyWithZ(this, zDestOffset)
+
+public function camerasetup.applyForceDuration(boolean doPan, real forceDuration)
+    CameraSetupApplyForceDuration(this, doPan, forceDuration)
+
+public function camerasetup.applyForceDurationWithZ(real zDestOffset, real forceDuration)
+    CameraSetupApplyForceDurationWithZ(this, zDestOffset, forceDuration)
+
+public function camerasetup.apply(boolean doPan, player whichPlayer, real duration)
+    CameraSetupApplyForPlayer(doPan, this, whichPlayer, duration)
+
+public function camerasetup.getFieldSwap(camerafield whichField) returns real
+    return CameraSetupGetFieldSwap(whichField, this)
+
+public function player.cameraSetupApply(boolean doPan, camerasetup whichSetup, real duration)
```

Comment:
I'm not really in love with some of these `player` extension-functions.

Reply:
If you want me to change anything, you will have to tell me what exactly bothers you about them.

Reply:
I don't think I'm in a position to demand changes, but I wonder if someone else agrees that `player` extension-functions that are specific to `camerasetup` or otherwise `camera` handling might belong in a separate package that is imported explicitly.

Reply:
I think it's fine as long as this package is an auto import, which it should be.

Reply:
eh, I feel like it should be camerasetup's extension func instead of player's

Reply:
Frotty my point about using a separate package imported explicitly is the opposite of what you just said.

In particular, I think `Player(3).panToTimed(vec2(0., 0.), 0.)` is weird to have on auto-import.

Reply:
@Trokkin That already exists. I made both versions (for this function and a few others). The one you are looking for is camerasetup.apply, where you specify for which player to do this.

@Cokemonkey11 The function you just mentioned doesn't seem to exist? player.panToTimed isn't in the file. The function is called player.panCameraToTimed, which to me makes sense: Player(3).panCameraToTimed(pos, duration) = For Player 3, pan camera to pos timed over duration.

Reply:
Just saying "it's weird" doesn't help, coke. I think it's totally fine, and I don't see what your problem is. When thinking naturally about this occuring in code, applying a camera position almost always happens together with other player related context. E.g. selecting a unit, adding some resources, etc. and in the same way I would expect to change the player's camera. 
Camerasetup is seldom used, and calling an extension func on it wouldn't allow for cascading. Though it should still be available to the developer, so I think having both is a good idea.
But perhaps they should rely on a shared function to ensure they remain the same.

Reply:
I didn't know anyone cared why I think it's weird :D actually I just mentioned it in case someone else felt the same.

The reason I think it's weird is that a `player` represents some abstract game/model related interface. Player's methods usually involve manipulating a player's properties in the context of the game. Camera and `camerasetup` is an abstraction around a *view* of the game playing field. Specifying a player when interacting with them only ever impacts the scope over which those views are manipulated.

In other words, I think of the built-in free-function camera API like:

- camera free functions are tools for changing a view over the game terrain
- camera free functions sometimes (always? does it matter?) take a `player` as a means for limiting that scope
- `player` in the sense of the model as related to the game is otherwise not impacted

Anyway, if I really hated this idea, I would be shouting, but I don't, so I just say "seems weird"